### PR TITLE
enable sling models and create a Module model

### PR DESCRIPTION
**Pull request by carlosmunoz**
Enable the sling models functionality on this application. Also create an initial version of a Module model to statically map all the properties of modules from our model.
Models help by putting all mapping information (from JCR to java classes) into a single place. If adopted, no code should use the JCR or sling resource apis going forward.

### CODE REVIEW

File: src/test/java/com/redhat/pantheon/servlet/AsciidocRenderingServletTest.java [hunk 1]
```diff
@@ -12,6 +14,7 @@
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
+import sun.misc.Cache;
```
[the remark this answers is not in the document]
Comment (carlosmunoz) [reply src/test/java/com/redhat/pantheon/servlet/AsciidocRenderingServletTest.java]: Yes, definitely a mistake.

File: pom.xml [hunk 2]
```diff
@@ -240,6 +250,18 @@
...
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.sling</groupId>
+            <artifactId>org.apache.sling.models.api</artifactId>
+            <version>1.3.6</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <artifactId>geronimo-atinject_1.0_spec</artifactId>
```
[the remark this answers is not in the document]
Comment (carlosmunoz) [reply pom.xml]: They are coming from this package. In fact I went into the Sling config (Felix) and found the exact bundles providing these classes and this is the one that showed up.

**Comment by carlosmunoz** [comment]
@benradey please review again? code corrected and answers provided :smile:

File: pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (benradey) [reply pom.xml]: Ok, sounds like you did your homework!

**Pull request merged**